src/porepy/models/compositional_flow.py

```diff
@@ -0,0 +1,2070 @@
+        ]
+
+    def set_equations(self) -> None:
+        """Set the equations for the mass balance problem.
+
+        A mass balance equation is set for all independent components on all subdomains.
+
+        """
+        super().set_equations()
+
+        subdomains = self.mdg.subdomains()
+
+        for component in self.fluid.components:
+            if self.has_independent_fraction(component):
+                sd_eq = self.component_mass_balance_equation(component, subdomains)
+                self.equation_system.set_equation(sd_eq, subdomains, {"cells": 1})
+
+    def component_mass_balance_equation(
+        self, component: pp.Component, subdomains: list[pp.Grid]
+    ) -> pp.ad.Operator:
+        """Mass balance equation for subdomains for a given component.
+
+        Parameters:
+            component: A component in the :attr:`fluid`.
+            subdomains: List of subdomains.
+
+        Returns:
+            Operator representing the mass balance equation.
+
+        """
+        # Assemble the terms of the mass balance equation.
+        accumulation = self.volume_integral(
+            self.component_mass(component, subdomains), subdomains, dim=1
+        )
+        flux = self.component_flux(component, subdomains)
+        source = self.component_source(component, subdomains)
+
+        # Feed the terms to the general balance equation method.
+        eq = self.balance_equation(subdomains, accumulation, flux, source, dim=1)
+        eq.set_name(self._mass_balance_equation_name(component))
+        return eq
+
+    def component_mass(
+        self, component: pp.Component, subdomains: list[pp.Grid]
+    ) -> pp.ad.Operator:
+        r"""Returns the accumulation term in a ``component``'s mass balance equation
+        using the :attr:`~porepy.compositional.base.FluidMixture.density` of the fluid
+        mixture and the component's
+        :attr:`~porepy.compositional.base.Component.fraction`
+
+        .. math::
+
+            \Phi \rho \z_{\eta},
+
+        in AD operator form on the given ``subdomains``.
+
+        Parameters:
+            component: A component in the :attr:`fluid`.
+            subdomains: A list of subdomains on which above operator is called.
+
+        Returns:
+            Above expression in AD operator form.
+
+        """
+        mass_density = (
+            self.porosity(subdomains)
+            * self.fluid.density(subdomains)
+            * component.fraction(subdomains)
+        )
+        mass_density.set_name(f"component_mass_{component.name}")
+        return mass_density
+
+    def advection_weight_component_mass_balance(
+        self, component: pp.Component, domains: pp.SubdomainsOrBoundaries
+    ) -> pp.ad.Operator:
+        """The non-linear weight in the advective component flux.
+
+        It uses the ``component``'s :meth:`~porepy.models.fluid_property_library.
+        FluidMobility.fractional_component_mobility`, assuming the flux contains the
```

Comment:
I checked the class, this is only documented here.

But I improved the documentation, to make it clear that it can be used for both fractional flow and standard formulation.